### PR TITLE
gip-0/burn-deployer-funds

### PR DESCRIPTION
**Pull request by empty-set-gold**
### Background

During the initial deployment of ESG, 0.5ESG was minted and bonded to the DAO. 
This was to ensure that in the very early days of ESG expansion, the creators retained a sizeable share of the DAO in the face of bots competing to call advance().

**This initial ESG now accounts for 43%~ of the DAO voting power**

Legitimate concerns have been raised with regards to the presence of this (almost majority) share of voting power, and as the 0.1ESG reward for calling advance() is now numerically insignificant in the face of 10% expansions of a much larger supply, it's time to change this allocation.

### Solutions
An outline of the discussed solutions can be found in the gip-φ channel of the Discord: https://discord.gg/JYgZtJwXAP
After some time, the following approach seemed to satisfy most people: 

> A combination of token/share burn, where the team retain a _maximum_ of 15% voting power, and also some method of ensuring that value is gradually released.

**With these rough requirements in mind, we would like to present the following approach:** 

1.  Burn 70% of deployer ESGS/ESG in place.  No unbonding required.
2.  Lock deployer funds in place until June 1st 2021.

### Why we like this approach

1. By locking funds within the protocol instead of an external service, the DAO retains ultimate power over those funds.  If June 1st approaches, and the community do not wish to release those funds for whatever reason, it is simply a matter of passing another vote to increase the deadline. 

2. By staying within the protocol, the deployer retains its voting power (approx 13%~) after the token burn, and this % meets the maximum requirement of 15% voting power.

3. Very few moving pieces with regard to managing those funds.  Less code is better, especially for something not directly related to the protocol itself.

### CODE REVIEW

File: protocol/contracts/dao/Bonding.sol [hunk 1]
```diff
@@ -77,4 +77,16 @@ contract Bonding is Setters, Permission {
...
+    function burnDeployerStake(uint256 percentage) internal {
+        address deployer = getDeployerAddress();
+        uint256 currentBalance = balanceOfBonded(deployer);
+        uint256 burnable = currentBalance.mul(percentage).div(100);
+        uint256 burnableShares = burnable.mul(totalSupply()).div(totalBonded());
+
+        // Deliberately not performing incrementBalanceOfStaged
+        decrementTotalBonded(burnable, "Bonding: insufficient total bonded");
+        decrementBalanceOf(deployer, burnableShares, "Bonding: insufficient balance");
+        gold().burn(burnable);
```
Comment (empty-set-gold) [inline protocol/contracts/dao/Bonding.sol:90]: Committer does not incur the burn here; msg.sender becomes the DAO in the external call to the `new Gold()` contract.

File: protocol/contracts/dao/Implementation.sol [hunk 2]
```diff
@@ -16,8 +16,8 @@ contract Implementation is State, Bonding, Market, Regulator, Govern {
     event Incentivization(address indexed account, uint256 amount);
 
     function initialize() initializer public {
-        // Dev rewards
-        incentivize(msg.sender, 5e17); // 0.5 ESG
```
Comment (empty-set-gold) [inline protocol/contracts/dao/Implementation.sol:20]: 🔥

File: protocol/contracts/dao/Bonding.sol (hunk 1 above)
Comment (andy8052) [reply protocol/contracts/dao/Bonding.sol:90]: I like this solution

**Review by andy8052 — APPROVED** [review]
lgtm!

**Pull request merged**